torchao/quantization/prototype/mixed_precision/scripts/BO_w_sensi_samples.py

```diff
@@ -0,0 +1,242 @@
+import torch
+import torch.nn as nn
+
+from torchao.quantization import quantize_
+
+import copy
+import random
+
+from lm_eval.evaluator import evaluate
+from lm_eval.models.huggingface import HFLM
+from lm_eval.tasks import get_task_dict
+
+from naive_intNwo import intN_weight_only
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from ax.service.ax_client import AxClient, ObjectiveProperties
+
+
+# load model
+device = "cuda" if torch.cuda.is_available() else "cpu"
+repo_id = "/home/hanxianhuang/ao/torchao/quantization/prototype/mixed_precision/checkpoints/meta-llama/Meta-Llama-3-8B"
+tokenizer = AutoTokenizer.from_pretrained(repo_id)
+model = AutoModelForCausalLM.from_pretrained(repo_id, torch_dtype=torch.bfloat16).to(
+    device=device
+)
+
+
+# quantize a model based on given configuration
+def quantize_by_fqn_to_config(model, fqn_to_config):
+    for fqn, config in fqn_to_config.items():
+        bit_width, groupsize = config.split("_")
+        bit_width = int(bit_width)
+        groupsize = int(groupsize)
+
+        def filter_fn_sen(child: torch.nn.Module, cur_fqn: str) -> bool:
+            return isinstance(child, torch.nn.Linear) and (fqn in cur_fqn)
+
+        quantize_(
+            model.to(device=device),
+            intN_weight_only(n=bit_width, group_size=groupsize),
```

Comment:
nit: we could be using https://github.com/pytorch/ao/pull/672 as well, that will make the model size calculation easy: https://github.com/jerryzh168/ao/blob/e06c05a432c74dea395ce40dd8454d25fed7fac2/test/dtypes/test_uintx.py#L122, although I feel we could have an option to remove zero_point as well